packages/hoprd/src/api/v2/logic/withdraw.ts

```diff
@@ -0,0 +1,69 @@
+import Hopr from '@hoprnet/hopr-core'
+import { Balance, moveDecimalPoint, NativeBalance } from '@hoprnet/hopr-utils'
+import { isError } from '.'
+
+type WithdrawArgs = {
+  amount: string
+  weiAmount: string
+  currency: 'NATIVE' | 'HOPR'
+  recipient: string
+}
+
+const validateWithdrawArgs = async ({
+  amount,
+  currency,
+  recipient
+}: {
+  amount: string
+  currency: string
+  recipient: string
+}): Promise<WithdrawArgs | Error> => {
+  const validCurrency = currency.toUpperCase() as 'NATIVE' | 'HOPR'
+  if (!['NATIVE', 'HOPR'].includes(validCurrency)) {
+    return new Error('incorrectCurrency')
+  }
+
+  if (isNaN(Number(amount))) {
+    return new Error('incorrectAmount')
+  }
+
+  // @TODO: validate recipient address
```

Comment:
can be done by using one of our utils (https://github.com/hoprnet/hoprnet/blob/731a29dd61ff81b678526992999737f2c31481d2/packages/utils/src/types/primitives.ts#L114) from `@hoprnet/hopr-utils`